cpp/src/inflight_requests.cpp

```diff
@@ -67,6 +72,38 @@ void InflightRequests::remove(const Request* const request)
   } while (true);
 }
 
+size_t InflightRequests::dropCanceled()
+{
+  size_t removed = 0;
+
+  {
+    std::scoped_lock lock{_cancelMutex};
+    for (auto it = _cancelingRequests->begin(); it != _cancelingRequests->end();) {
+      auto request = it->second;
+      if (request != nullptr && request->getStatus() != UCS_INPROGRESS) {
+        it = _cancelingRequests->erase(it);
```

Comment:
If `getStatus` is not `UCS_OK` (and not `UCS_INPROGRESS`) do we need to throw an exception, or is that handled elsewhere?

Reply:
No, this is all handled by `ucxx::Request`, here we're only tracking what to do with its internal reference, preventing the object from going out-of-scope and causing a segfault when the UCP request callback is executed if the user released it too early.